TopMovies/Core/Action/MoviesListAction.swift

```diff
@@ -0,0 +1,13 @@
+//
+//  MoviesListAction.swift
+//  TopMovies
+//
+//  Created by anikolaenko on 02.02.2021.
+//
+
+import ReSwift
+
+enum MoviesListAction: Action {
+    case getMovies
+    case saveMovies(movies: [MovieModel]?)
+}
```

Comment:
How about this one:
```swift
enum MoviesListAction {
  case request
  case downloading
  case completed(movies: [Movie])
  case failed(error: Error)
}
```

Reply:
I'm really confused about "downloading" action, is it really needed here? Maybe i should just drop it while MoviesListState changes to "downloading" state in response to "request" action? :]

Reply:
State `requested` shows that someone requested a movies list. But it's not a guarantee that the service has started loading. It may ignore it due to some circumstances. So `downloading` action will be dispatched right before the service will start a request.

Reply:
Thank you!